### DISCUSSION TITLE
Allow embedding via iframe

### DISCUSSION THREAD

**Discussion by chris-eaton** [GitbookIO/community#39] [Feature Requests]
### What's your feature/idea?

AS A paid subscriber
I WANT to embed GitBook pages and collections within my website
SO THAT I can present a seamless experience to my end users e.g. they dont feel like they are leaving my site to goto a help articles

---

I can understand blocking iframe embeds from cross domains, however in my case I have gitbook setup to use my own domain, but I cannot embed iframes on either those domains or via localhost (required for development)

### What's your desired outcome?

As above I want to embed my gitbook pages into my webpage via iframe

### What's the impact of this for you?

- [X] Critical
- [ ] High
- [ ] Medium
- [ ] Low

**Comment by ShutdownRepo** [comment]
The request feels a bit aggressive tbh, but I agree that iframes would be a nice touch

**Reply by chris-eaton** [reply]
Hi @ShutdownRepo apologies if it came across that way, it was not intentional. I tried to use standard product terminology as I often find natural language requests can cause confusion.

**Comment by SamyPesse** [comment]
I thought it was already supported. For example with our own docs:

<img width="952" alt="Screenshot 2023-01-30 at 12 02 54" src="https://user-images.githubusercontent.com/845425/215459870-182e9709-2ddc-4df3-9cbc-703eeed93a02.png">

Can you share the link to your documentation so that I can run some tests?

**Reply by ShutdownRepo** [reply]
I think @chris-eaton's request was a bit different than mine and I think his scenario would indeed work. 
On my end, I was thinking the other way around. I'd like the ability to include iframes from external content within a gitbook page

**Reply by SamyPesse** [reply]
@ShutdownRepo Makes sense! What iframe would you like to insert in GitBook? What is the underlying service?

We are planning on opening access to our integrations platform, where developers will be able to build custom blocks, making it possible to embed iframes from different sources.

**Reply by ShutdownRepo** [reply]
I was thinking about integrating Google AdSense.
Google Analytics and Search Console already work, but not adsense. Since I'm running an open-source project that demands a huge amount of work, AdSense would be one of the only ways I know to keep the project going at full speed without charging any money from its users.

**Reply by ShutdownRepo** [reply]
With that being said, I'm really looking forward to the ability to customize integrations, to integrate AdSense, Discord webhooks, Twitter etc.

**Reply by chris-eaton** [reply]
Hi @SamyPesse Here is an example link.  https://docs.verilocation.com/api-documentation/

Here is how the code is embedded (currently viewed from a page on localhost)
``` <iframe height="300" width="300" src="https://docs.verilocation.com/api-documentation/"></iframe>```

The error shown in the console is:
> Refused to frame 'https://docs.verilocation.com/' because an ancestor violates the following Content Security Policy directive: "frame-ancestors 'self' https: https://* http://*".

**Reply by SamyPesse** [reply]
@chris-eaton, maybe it's only happening in `localhost`. The `frame-ancestors` should allow the documentation to be embedded in any `http://` or `https://` website.

It might also be related to the security header you are serving for `localhost`, you might have a Content-Security-Policy that prevents the loading of any iframe.

I know some other customers who are already doing it.

**Comment by addisonschultz** [comment]
Hi @chris-eaton @ShutdownRepo!

Addison here from GitBook - we’re working on a few things related to this feature request :)

Can you reach out to me at addison (@) gitbook.com with the Subject on “Embeddable GitBook”?

Would love to chat more!
